src/exceptions.rs

```diff
@@ -743,7 +743,7 @@ macro_rules! test_exception {
 
                 assert!(err.is_instance_of::<$exc_ty>(py));
 
-                let value: &$exc_ty = err.value(py).downcast().unwrap();
+                let value: &$exc_ty = err.value_bound(py).into_gil_ref().downcast().unwrap();
```

Comment:
I couldn't work out if we can avoid `into_gil_ref()` here.

Reply:
I think the what leads to that here is that `Bound<PyException>` probably shouldn't implement `std::error::Error`, because we can't implement `.source()`; that relied on the way that GIL Refs allowed abusing reference ownership.

We should add a point into the migration guide that `Bound<PyException>` (for all exception types) no longer implements `std::error::Error`, similar to how `Bound<PyList>` no longer implements `std::ops::Index.

So yes, we can leave the conversion to gil ref here.